lib/elixir/lib/kernel/parallel_compiler.ex

```diff
@@ -62,7 +62,10 @@ defmodule Kernel.ParallelCompiler do
       the file, module and the module bytecode
 
     * `:each_cycle` - after the given files are compiled, invokes this function
-      that return a list with potentially more files to compile
+      that should return the following values:
+      * `{:compile, modules}` - to continue compilation with a list of further modules to compile
+      * `{:runtime, modules}` - to stop compilation and verify the list of modules because
```

Comment:
Why is this called `:runtime`? 🙂 

Maybe `:check` would be better as you might have compile time references in dependent modules?

Reply:
I wasn't sure about the names. The name comes from compile-time dependent modules and runtime dependent modules. They are all checked (because we check all compiled/changed modules and runtime dependent modules) so I don't think `:check` would be a great name either.

Reply:
I interpreted it as the next action to perform after each cycle: compile or check.

Reply:
WDYT @josevalim?

Reply:
Any approach works to me. :+1: The thing to note is that we go to `:check` even if you don't return the `:check` tuple.